### ISSUE TITLE
Maui -Android certificate pinning using network-security-config ignore incorrect pin

### ISSUE THREAD

**Issue by DmitryNikiforovCoherent** [dotnet/maui#24624]
### Description

Added ssl pinning using network-security-config

For test change one letter of base 64 pin, but SslClientAuthenticationOptions.RemoteCertificateValidationCallback still pass SslPolicyErrors.None

### Steps to Reproduce

-  Added network_security_config.xml as AndroidResource `<network-security-config>

    <domain-config>
        <domain includeSubdomains="true">{host}</domain>
        <pin-set expiration="2025-03-21">
            <pin digest="SHA-256">pin</pin>
        </pin-set>
    </domain-config>
</network-security-config>` 

- Added SocketsHttpHandler with SslClientAuthenticationOptions.RemoteCertificateValidationCallback
- I expect, that if I change base64 pin to incorrect, RemoteCertificateValidationCallback will be called with RemoteCertificateNameMismatch, as for IOS, but receive SslPolicyErrors.None

### Version with bug

8.0.70 SR7

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Any android version

**Comment by QianaJiao** [comment]
@DmitryNikiforovCoherent Could you share us a repo, I'd like to verify it.

**Comment by RobTF** [comment]
We have the thumbprint in app config and use [`ServerCertificateCustomValidationCallback`](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclienthandler.servercertificatecustomvalidationcallback?view=net-8.0) and that seems to work on all platforms.

Is there a reason certificate pinning has been special cased into a separate config file etc.?

**Comment by DmitryNikiforovCoherent** [comment]
@QianaJiao try to prepare demo soon. Just one note:
Trying 2 httpclient options

1. HttpClientHandler with ServerCertificateCustomValidationCallback, any sha256 pin(even correct) throw RemoteCertificateChainErrors, for all chains(try first certificate, intermediate and CA)
2. SocketsHttpHandler with SslClientAuthenticationOptions.RemoteCertificateValidationCallback always returns SslPolicyErrors.None, even if pin incorrect
At the same time, ios works correctly for both cases
Could you verify, maybe I do smth wrong?
```
<?xml version="1.0" encoding="utf-8"?>
<network-security-config>
    <domain-config>
        <domain includeSubdomains="true">{host}</domain>
        <pin-set>
            <pin digest="SHA-256">Base 64 pin</pin>
        </pin-set>
    </domain-config>
</network-security-config>

```
Base 64 pin I receive from [ssllabs](https://www.ssllabs.com/) for my host
![image](https://github.com/user-attachments/assets/de019bc2-3bf2-432d-9503-41a3b4faad8e)

**Comment by QianaJiao** [comment]
Thank you for your reply. I will continue to investigate after I get your demo.
In addition, have you noticed these 2 points?

1. Ensure that the build action of the network_security_config.xml file is set to AndroidResource
![image](https://github.com/user-attachments/assets/039aca23-cf3a-463b-a2ae-f5742381826b)
2. Configure the networkSecurityConfig property on the application node in the Platforms\Android\AndroidManifest.xml file in your .NET MAUI app project

Here are some documents that may also be useful to you
https://learn.microsoft.com/en-us/dotnet/maui/data-cloud/local-web-services?view=net-maui-8.0#enable-clear-text-network-traffic-for-the-localhost-domain
https://developer.android.google.cn/privacy-and-security/security-config

**Comment by DmitryNikiforovCoherent** [comment]
@QianaJiao 
Yes, Build action is AndroidResource, network_security_config is mentioned in AndroidManifest.
Could you check this? HttpClientService contains both cases, with HttpClientHandler and SocketsHttpHandler, please set breakpoints on Line 32 and Line 81 to check SslPolicyErrors, they will call immediately after start.
network_security_config.xml contains correct and incorrect pin, please comment what you need
I use api from https://api-ninjas.com/api/exercises, receive pins from https://www.ssllabs.com/ssltest/analyze.html?d=api.api%2dninjas.com&s=3.168.86.93&latest

Sorry for non user friendly sample, please notify if you have any questions

[Archive.zip](https://github.com/user-attachments/files/16942169/Archive.zip)

**Comment by QianaJiao** [comment]
I can repro this issue on both Android and iOS. No matter which `pin` I comment, it will return SslPolicyErrors.None. 
I'm not confident about my verification results. Could anyone help to take a look？

**Comment by DmitryNikiforovCoherent** [comment]
@QianaJiao  Sorry, on ios I haven't added ssl pinning, just on android(As ios works correct for me, haven't touched in demo). On android have you checked both button clicks?
<img width="401" alt="image" src="https://github.com/user-attachments/assets/fe7995a3-0748-4250-9a43-109bf9a69d0f">
Socket Handler always works successfully, Http Handler always returns error

**Comment by jfversluis** [comment]
@jonathanpeppers is this Android or even runtime?

**Comment by jonathanpeppers** [comment]
There is a bit of confusion on this thread, so I'm not sure I follow what the problem is. Is there a sample that displays a problem?

One note:
* Using `AndroidMessageHandler` (default), probably no `RemoteCertificateValidationCallback` would ever be called. It would be using the native Android http stack in that case.
* Using `SocketsHttpHandler` (you'd have to set `UseNativeHttpHandler=false` in your `.csproj`), `RemoteCertificateValidationCallback` should get called

@simonrozsival might be able to help, if we think there is a runtime issue here.
